### ISSUE TITLE
Make backlight / alsa bar clickable

### ISSUE THREAD

**Issue by Quoteme** [polybar/polybar#2562]
## Is your feature request related to a problem? Please describe.
I would like the bar from the backlight module to respond to where I click on it. If I click on the left edge of the bar, I would like my backlight to be set to 0%, if I click on the right edge, I would like my backlight at 100% and if I click at the center, I would like my backlight to be at 50%. Same for the volume bar.

## Why does polybar need this feature?
This greatly benefits people on a touchscreen monitor.

## Describe the solution you'd like
I provided a solution in my first paragraph.

## Describe alternatives you've considered
xmobar. I wrote a module for xmobar myself which does exactly that. But xmobar cannot handle my systemtray.

**Comment by lucax88x** [comment]
And if I may, possibily, allow click-left click-middle click-right like other modules.

**Comment by Quoteme** [comment]
yeah, maybe it would be a good idea to have some options like `click-left-10% = backlight --set 10%`, `click-left-20% = backlight --set 20%` and so on?

**Comment by patrick96** [comment]
This has crossed my mind several times already. If anyone wants to work on this feel free :)

It's going to be a bit of a pain to get this into the code because the progressbar code doesn't know anything about click actions and it requires adding an action tag around every segment of the progressbar. In addition this will require new [module actions](https://polybar.readthedocs.io/en/stable/user/actions.html) for all modules with clickable progressbars (alsa, pulseaudio, xbacklight, and backlight. mpd also has such an action).

> yeah, maybe it would be a good idea to have some options like click-left-10% = backlight --set 10%, click-left-20% = backlight --set 20% and so on?

I don't think that's a good idea. For one, the percentages are not always integers and depend on the size of the progressbar. Also, polybar already has ways to change the properties displayed by the progressbars (scrolling on the modules already does this).

@lucax88x There is already an issue for this #1524

**Comment by xoores** [comment]
I added clickable feature in backlight module in #2703 - it toggles between min & max however 50% could be added as a step also...

**Comment by Quoteme** [comment]
> I added clickable feature in backlight module in #2703 - it toggles between min & max however 50% could be added as a step also...

That definitely is a step in the right direction!

**Comment by xoores** [comment]
It's waiting for merge :slightly_smiling_face:

I might have another (perhaps more elegant) idea for your usesace: I could add left/middle/right click to set 0/50/100% - depending on touchscreen config it might b one/two/three finger tap on the module to set it to the corresponding value. How about that?

**Comment by xoores** [comment]
I don't think this is easily achievable unfortunately and I don't think this will get merged - my old PR was already in the "toddler" stage and I think it would start walking by now :laughing: 

I'm currently looking for an alternative bar as my old fork randomly freezes and current version has no `reverse-scroll` option for backlight & volume :sweat_smile: I'm also looking into the Wayland (Hyprland + AGS looks sweet!) in the little free time I have.
